### DISCUSSION TITLE
Port to Steamworks SDK

### DISCUSSION THREAD

**Discussion by foxing-quietly** [m4dEngi/RemotePlayWhatever#181] [General]
I've been interested in this project for a while, including the possibility of integrating it as a decky loader plugin. If @m4dEngi or anyone else would like to reach out to me to help better understand the codebase, I'd love to contribute where I can.

I noticed that RemotePlayWhatever uses a [locally-maintained fork](https://github.com/m4dEngi/open-steamworks/tree/ae60ff362231f746d7ac79618714530053866e65). of open-steamworks. The [original](https://github.com/SteamRE/open-steamworks) open-steamworks is archived and considered abandoned since 2023 now that the official Steamworks SDK is publicly available.

I'm new to the Steamworks API and I fancy myself more of a Rust than C++ developer, but I was wondering if there is any functionality from open-steamworks you use that you can't get from the official SDK? Is there any technical reason RemotePlayWhatever can't be ported over to the official SDK?

**Comment by m4dEngi** [comment]
It uses an undocumented, unstable API not accessible in the official Steamworks SDK, so OSW.

**Reply by foxing-quietly** [reply]
I see. I'm still open to contributing to the existing codebase, is there any sort of additional documentation you can point me to for understanding either RemotePlayWhatever or the fork of open-steamworks?

**Reply by m4dEngi** [reply]
No, there's no documentation atm. The API is still undocumented and unstable.
